scripts/aggregate-themes/theme-aggregator.js

```diff
@@ -156,6 +91,84 @@ class ThemeAggregator {
     return assets;
   }
 
+  /**
+   * Triggers aggregation and generation for a given theme.
+   * @param {string} themeName - The theme to aggregate.
+   * @param {Object} options - The aggregation options.
+   * @param {boolean} defaultFlag - Whether the theme to be generated is a root or scope theme. Guards against default theme and scope theme being equivalent.
+   * @returns {array} - An array of file names.
+   */
+  static triggerAggregationAndGeneration(themeName, options = {}, defaultFlag) {
+    if (!themeName) return null;
+    Logger.log(`Aggregating ${themeName} files...`, { LOG_CONTEXT });
+    const { theme, scoped = [] } = options;
+    const isRoot = themeName === theme && defaultFlag;
+    const isScoped = scoped.indexOf(themeName) > -1;
+    const themeScope = { isRoot, isScoped };
+
+    const aggregatedAssets = ThemeAggregator.aggregateTheme(themeName, themeScope, options);
+    const generatedAsset = ThemeAggregator.generateTheme(themeName, themeScope, options);
+    // Generated theme file should take precedence over aggregated files.
+    // Therefore, take advantage of css import precdence by adding the generated asset last.
+    if (generatedAsset) {
+      aggregatedAssets.push(generatedAsset);
+    }
+
+    if (!aggregatedAssets.length) {
```

Comment:
Should this if check mach the `generatedAsset` if check?

Reply:
no, we want an existence check on all assets to correctly say whether no theme files were found, regardless if they were existing root/scoped or generated root/scoped files